chains/orchestrator-relays/runtime/dancelight/src/bridge_to_ethereum_config.rs

```diff
@@ -236,6 +242,179 @@ where
     }
 }
 
+/// `EthTokensLocalProcessor` is responsible for receiving and processing the ETH native
+/// token and ERC20s coming from Ethereum with Tanssi chain as final destination.
+pub struct EthTokensLocalProcessor<T, XcmProcessor, XcmWeigher, EthereumLocation, EthereumNetwork>(
+    PhantomData<T>,
+    PhantomData<XcmProcessor>,
+    PhantomData<XcmWeigher>,
+    PhantomData<EthereumLocation>,
+    PhantomData<EthereumNetwork>,
+);
+impl<T, XcmProcessor, XcmWeigher, EthereumLocation, EthereumNetwork> MessageProcessor
+    for EthTokensLocalProcessor<T, XcmProcessor, XcmWeigher, EthereumLocation, EthereumNetwork>
+where
+    T: snowbridge_pallet_inbound_queue::Config
+        + pallet_ethereum_token_transfers::Config
+        + pallet_foreign_asset_creator::Config,
+    T::AccountId: From<[u8; 32]>,
+    XcmProcessor: ExecuteXcm<T::RuntimeCall>,
+    XcmWeigher: WeightBounds<T::RuntimeCall>,
+    EthereumLocation: Get<Location>,
+    EthereumNetwork: Get<NetworkId>,
+{
+    fn can_process_message(channel: &Channel, envelope: &Envelope) -> bool {
+        // Ensure that the message is intended for the current channel, para_id and agent_id
+        if let Some(channel_info) =
+            pallet_ethereum_token_transfers::CurrentChannelInfo::<Runtime>::get()
+        {
+            if envelope.channel_id != channel_info.channel_id
+                || channel.para_id != channel_info.para_id
+                || channel.agent_id != channel_info.agent_id
+            {
+                return false;
+            }
+        } else {
+            return false;
+        }
+
+        // Check it is from the right gateway
+        if envelope.gateway != T::GatewayAddress::get() {
+            return false;
+        }
+
+        // Try decoding the message
+        match VersionedXcmMessage::decode_all(&mut envelope.payload.as_slice()) {
+            Ok(VersionedXcmMessage::V1(MessageV1 {
+                command:
+                    Command::SendToken {
+                        token: token_address,
+                        destination: Destination::AccountId32 { id: _ },
```

Comment:
Do I assume that destination will be different in the case of sending it to a para-id? does this mean that it could be expanded later on? if so I would try put common code between `can_process` and `process`.